internal/server/commonpool.go

```diff
@@ -0,0 +1,31 @@
+package server
+
+import (
+	"math"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+//islandDistribute distributes resources to island as requested
+func (s *SOMASServer) islandDistribute(resource shared.Resources, islandID shared.ClientID) {
+	ci := s.gameState.ClientInfos[islandID]
+	ci.Resources = ci.Resources + resource
+	s.gameState.ClientInfos[islandID] = ci
+	s.gameState.CommonPool -= resource
+}
+
+//islandDeplete depletes island's resource based on the severity of the storm
+func (s *SOMASServer) islandDeplete(porportionalEffect map[shared.ClientID]float64) {
+	for id, ci := range s.gameState.ClientInfos {
+		ci.Resources = shared.Resources(math.Max(float64(ci.Resources)-porportionalEffect[id], 0)) // min resources = 0
+		s.gameState.ClientInfos[id] = ci
+	}
+}
+
+// islandContribute takes resource donation from individual island into the common-pool
+func (s *SOMASServer) islandContribute(resource shared.Resources, islandID shared.ClientID) {
+	ci := s.gameState.ClientInfos[islandID]
+	ci.Resources = ci.Resources - resource
+	s.gameState.ClientInfos[islandID] = ci
+	s.gameState.CommonPool += shared.Resources(resource)
+}
```

Comment:
Where would be most logical to include this islandContribute method to request CP contributions from clients?

Reply:
The current location is in IIGOTax, while the name suggests its for tax we can repurpose it a make it so there's only a single point in the turn where agents contribute to the CP. Both for tax and for disaster mitigation purposes

Reply:
Great, thanks @NGovani! And should we still have two distinct contribution requests within IIGOTax (one for tax, one for CP)?

Reply:
Since if we do just one request for tax and then allocate (contrib - tax) surplus to CP, how could agents choose to not pay tax but still contribute to CP?

Reply:
Hmm thats a good point, I think we can just say that you can by happy accident you pay your tax because you want to contribute to the CP. I doubt an agent which is complicated enough to somehow protest IIGO but still pay to the common pool for the greater good will be created

Reply:
Yeah I agree that it's unlikely, but why would we not just want to have two separate functions to allow for the possibility? It's not like it really complicates things at all? Have already implemented in this PR. Can change back though if there's a strong reason why this wouldn't be a good idea

Reply:
Oh if you've implemented it already then that's absolutely fine.  No issues here

Reply:
Sure thing 👍